### PR TITLE
move circle.yml code into external scripts

### PR DESCRIPTION
**Pull request by wjordan**
this allows this CI-build logic to be updated by the post-checkout `merge-base` command when modified on staging.

Haven't thoroughly tested this- putting it here for discussion, to see what people think of this approach and if it could solve our current issue with our post-checkout merge process on CircleCI getting out of sync on feature branches.

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -20,21 +20,10 @@ machine:
...
-    # be >= 52.0.2743.0
-    - google-chrome --version
-    - ./bin/circle/upgrade-google-chrome.sh
-    - google-chrome --version
-    # Install Yarn
-    - sudo apt-key adv --keyserver pgp.mit.edu --recv D101F7899D41F3C3
-    - echo "deb http://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
-    - sudo apt-get update -qq
-    - sudo apt-get install -y -qq yarn
+    - ./bin/circle/dependencies_pre.sh
```
Comment (pcardune) [inline circle.yml:26]: I would add a comment here explaining why we do this.

**Review by pcardune — COMMENTED** [review]
One downside to this approach that I can think of is that it doesn't show up in the circle ci output. But perhaps if we make the bash script echo out each command, that would be good enough.

File: circle.yml [hunk 2]
```diff
@@ -43,70 +32,14 @@ dependencies:
...
-          RAKE_VERBOSE: true
-        timeout: 840
-    # Run rake build (twice in case of intermittent test failures within rake build)
-    # See notes above rake install for timeout, ts, and mispipe reasoning
-    - for i in 1 2; do mispipe "bundle exec rake build" "ts '[%Y-%m-%d %H:%M:%S]'" && break; done:
-        timeout: 840
-    # Update firebase security rules and config params. If these security rules change in a
-    # non-backward-compatible way, this may cause a new test run to make an older one fail.
-    - rake firebase:ci
+    - ./bin/circle/dependencies_override.sh
```
Comment (bcjordan) [inline circle.yml:35]: Will want to keep a `timeout: 840` here to avoid Circle from shutting it down if no output for > N minutes

**Review by bcjordan — APPROVED** [review]

**Comment by pcardune** [comment]
is this PR dead?

**Comment by wjordan** [comment]
pretty sure this is still a relevant approach to a known issue, but it's low on my personal priorities to finish thorough testing, assume responsibility for any related issues if it breaks CI post-merge, etc.

if this PR is relevant to you or anyone else's current work, feel free to take it over and move it forward. it's probably not something i'm going to be spending additional time on in the near future.

**Comment by islemaster** [comment]
Hey @wjordan! Since we've started using Drone, I'm going to close this PR out - I don't think it's something we'll want to resurrect in this form (although it's potentially still a useful reference).  FYI @uponthesun in case you have thoughts about this work in the context of our Drone configuration.

**Pull request closed without merging**